Merge and Normalize K-mer Count Tool
This Python script, merge_normalize_isoform_count_v1.py, merges k-mer count data with original k-mer CSV files and adds normalized k-mer counts for bioinformatics analysis. It is designed to work with data generated from high-throughput sequencing experiments.

Overview
The script fulfills two main functions:

Merging k-mer count files generated by a k-mer counting tool with the original k-mer CSV files to centralize the data.
Normalizing the merged k-mer counts against the total number of reads, number of k-mers, the read length, and the k-mer size to account for sequencing depth and k-mer length bias.
Normalization is essential for comparing k-mer frequencies across samples or datasets with different sizes or sequencing depths.

Features
Merging of k-mer count CSV files with original k-mer CSV files on the 'kmer' column.
Normalization of k-mer counts to enable fair cross-sample comparisons.
Automated calculation of the total number of reads from a gzipped FASTQ file using the Bio.SeqIO module.
Support for processing multiple k-mer CSV files within a specified directory.
Easy specification of the target directory, output directory, FASTQ file, read length, and k-mer length through command-line arguments.

Usage
To use the script, provide the required arguments for input and output directories, the gzipped FASTQ file, and optional read length and k-mer size. Here's an example command:
python merge_normalize_isoform_count_v1.py --directory /path/to/kmer_files --output_directory /path/to/output --fastq /path/to/fastq_file.fastq.gz --read_length 150 --k 50

--directory: The path to the directory with k-mer and k-mer counts CSV files.
--output_directory: The path to the directory where the merged and normalized CSV files will be saved.
--fastq: The path to the specific FASTQ.GZ file to calculate the total number of reads.
--read_length: (Optional) The read length of FASTQ sequences, defaults to 150.
--k: (Optional) The k-mer length, defaults to 50.

Output
The script will output CSV files in the specified directory with the following columns:

Original columns from the k-mer CSV files.
Count: Raw count of each k-mer from the k-mer count CSV file.
Normalized_K-mer_Count: Normalized count of each k-mer based on the total reads and other user-provided parameters.
Each output CSV file will be named according to the original k-mer CSV file, appended with _merged_normalized.
