GaussF Pipeline Usage Example
Overview
This section provides a step-by-step example of how to perform k-mer count analysis using sequencing data with the GaussF pipeline. The pipeline includes scripts for generating k-mer lists from isoform transcripts, counting k-mer occurrences in FASTQ files, normalizing these counts, and estimating abundance using Gaussian fitting (GaussF).

The transcript CSV files, which contain isoform transcript FASTA data, resulting k-mer lists, sequencing counts of k-mers, and GaussF quantified abundance are available in the gaussf folder within this repository.

Running the Pipeline
Here's how you can run the individual scripts of the GaussF pipeline:

1. k-mer Frequency Distribution
To list kmers corresponding to the minimal shared region of isoform transcripts for each isoform, execute the following command in your terminal:
python kmer_frequency_distribution_mini_shared.py --input /path/to/mart_export_ACTB.txt --output /path/to/ACTB_reference/
The specified CSV file in the directory ACTB_reference will be used to output the k-mer lists for each isoform.

2. k-mer Counting and Normalization

2.1 Counting k-mer Occurrences
Run the following command to count k-mer occurrences in a FASTQ file and normalize these occurrences to get RPKM:

./kmer_counter --kmer_dir /path/to/kmer_csv_dir --fastq_file /path/to/sequence.fastq.gz --output_dir /path/to/output_csv_dir --threads 4
![image](https://github.com/QiangSu/GaussF/assets/30870950/3c32974f-c39f-4440-9bd3-9f72113d4af7)

python kmer_counting_loop.py --k 50 --threads 30 --chunk_size 10000000 --fastq /path/to/data.fastq.gz --kmer_dir
/path/to/ACTB_reference --output /path/to/ACTB_reference
Note that the k-mer directory should be the same as the output directory.

2.2 Normalizing Counted k-mer Occurrences
Execute the script below to normalize counted k-mer occurrences:
python merge_mormalize_isoform_count_v1.py --directory /path/to/ACTB_reference --fastq /path/to/data.fastq.gz --output_directory /path/to/ACTB_reference/merge_data --read_length 150 --k 50

3. Quantifying Abundance with GaussF
Run the following command to perform the quantified abundance estimation:
python pipeline_abundance_GaussF_esti_loop.py --threshold 5 â€“input /path/to/ACTB_reference/merge_data --output /path/to/ACTB_reference/merge_data/results_file.txt

Example Data
The kmer_counting_loop.py script is part of the GaussF pipeline, which is designed to provide unbiased total count estimation for isoforms. The pipeline utilizes k-mer counting and normalization techniques to generate a comprehensive dataset for accurate abundance estimation.
In the gaussf/example_GaussF folder, an ACTB.txt file is provided, containing all current 23 isoforms. This fasta file serves as input for the kmer_frequency_distribution_mini_shared.py script. The output is an isoform-specified kmers.csv file, which is then used as input for the kmer_counting_loop.py script. This script generates new isoform-specified CSV files, which are further utilized for normalization.
The normalized CSV files obtained from the previous step are then processed by the pipeline_abundance_GaussF_esti_loop.py script to obtain the result.txt file. All the output files generated by these scripts are available in the same folder and ready to be used with the GaussF pipeline.

These results, which have been compiled in the "ACTB_isoform_GaussF_results_file.txt" file, are the product of meticulous analysis and calculations.

The GaussF pipeline provides a comprehensive dataset that ensures an unbiased total count estimation for isoforms, facilitating accurate and reliable abundance analysis.
Output Results
The output of the scripts has been uploaded to this repository for reference purposes. The results file allows users to compare and verify the output generated by running the GaussF pipeline with the provided example data.
Feel free to explore the example_GaussF folder to understand the structure and content of the results.
